lib/commander.ts

```diff
@@ -57,21 +57,46 @@ Commander.prototype.getBuiltinCommands = function () {
  */
 Commander.prototype.createBuiltinCommand = function (commandName) {
   return {
-    string: generateFunction(commandName, "utf8"),
-    buffer: generateFunction(commandName, null),
+    string: generateFunction(null, commandName, "utf8"),
+    buffer: generateFunction(null, commandName, null),
   };
 };
 
+/**
+ * Create add builtin command
+ *
+ * @param {string} commandName - command name
+ * @return {object} functions
+ * @public
+ */
+Commander.prototype.addBuiltinCommand = function (commandName) {
```

Comment:
Why this new function?

Reply:
This is not really necessary for this PR, but whilst looking into the code/ previous issued i noticed #621 and realised similar issue probably exists for commands created via createBuiltinCommand, this would be a possible fix for that also but think it  would require https://github.com/luin/ioredis/blob/master/lib/pipeline.ts#L64 to be change to:
```typescript
Pipeline.prototype = Object.create(Commander.prototype);
```
for this to actually work.

Reply:
Got it about the new function.
Why replacing the prototype rather than merging it?

Reply:
👍

Reply:
Ended up doing this slightly differently as ran into issues with Redis inheriting from EventEmitter. I saw that defineCommand was doing a similar thing that i wanted to do so have done a similar implementation by adding the command to the instance rather than the prototype, then maintaining a set for the added commands.

Reply:
Wonderful. Much cleaner i think!